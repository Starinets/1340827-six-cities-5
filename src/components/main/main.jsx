import React from 'react';
import * as Type from '../../types';
import {NavLink} from "react-router-dom";

import OfferList from './../offer-list/offer-list';
import Map from './../map/map';

const Main = (props) => {

  const {
    rentalOffersCount,
    currentCity
  } = props;

  return (
    <div className="page page--gray page--main">
      <header className="header">
        <div className="container">
          <div className="header__wrapper">
            <div className="header__left">
              <NavLink className="header__logo-link header__logo-link--active" to="/">
                <img className="header__logo" src="img/logo.svg" alt="6 cities logo" width="81" height="41" />
              </NavLink>
            </div>
            <nav className="header__nav">
              <ul className="header__nav-list">
                <li className="header__nav-item user">
                  <NavLink className="header__nav-link header__nav-link--profile" to="/">
                    <div className="header__avatar-wrapper user__avatar-wrapper">
                    </div>
                    <span className="header__user-name user__name">Oliver.conner@gmail.com</span>
                  </NavLink>
                </li>
              </ul>
            </nav>
          </div>
        </div>
      </header>

      <main className="page__main page__main--index">
        <h1 className="visually-hidden">Cities</h1>
        <div className="tabs">
          <section className="locations container">
            <ul className="locations__list tabs__list">
              <li className="locations__item">
                <NavLink className="locations__item-link tabs__item" to="/">
                  <span>Paris</span>
                </NavLink>
              </li>
              <li className="locations__item">
                <NavLink className="locations__item-link tabs__item" to="/">
                  <span>Cologne</span>
                </NavLink>
              </li>
              <li className="locations__item">
                <NavLink className="locations__item-link tabs__item" to="/">
                  <span>Brussels</span>
                </NavLink>
              </li>
              <li className="locations__item">
                <NavLink className="locations__item-link tabs__item tabs__item--active" to="/">
                  <span>Amsterdam</span>
                </NavLink>
              </li>
              <li className="locations__item">
                <NavLink className="locations__item-link tabs__item" to="/">
                  <span>Hamburg</span>
                </NavLink>
              </li>
              <li className="locations__item">
                <NavLink className="locations__item-link tabs__item" to="/">
                  <span>Dusseldorf</span>
                </NavLink>
              </li>
            </ul>
          </section>
        </div>
        <div className="cities">
          <div className="cities__places-container container">
            <section className="cities__places places">
              <h2 className="visually-hidden">Places</h2>
              <b className="places__found">{ rentalOffersCount } places to stay in { currentCity }</b>
              <form className="places__sorting" action="#" method="get">
                <span className="places__sorting-caption">Sort by</span>
                <span className="places__sorting-type" tabIndex="0">
                Popular
                  <svg className="places__sorting-arrow" width="7" height="4">
                    <use xlinkHref="#icon-arrow-select"></use>
                  </svg>
                </span>
                <ul className="places__options places__options--custom">
                  <li className="places__option places__option--active" tabIndex="0">Popular</li>
                  <li className="places__option" tabIndex="0">Price: low to high</li>
                  <li className="places__option" tabIndex="0">Price: high to low</li>
                  <li className="places__option" tabIndex="0">Top rated first</li>
                </ul>
              </form>

              <OfferList
                offers = { props.offers }
                onCardMouseEnter = { props.onCardMouseEnter }
              />

            </section>
            <div className="cities__right-section">
              <Map
                offers = { props.offers }
              />
            </div>
          </div>
        </div>
      </main>
    </div>
  );
};

Main.propTypes = {
  offers: Type.OFFERS.isRequired,
  rentalOffersCount: Type.COUNT.isRequired,
  currentCity: Type.CITY.isRequired,
  onCardMouseEnter: Type.FUNCTION.isRequired
};

export default Main;
