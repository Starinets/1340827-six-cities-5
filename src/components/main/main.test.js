import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import {Main} from './main';
import {
  City,
  emptyFunction,
  SortList,
  Offers,
} from '../../test-data/test-data';

describe(`Component rendered correctly -> Main`, () => {
  it(`with offers`, () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(
        <Main
          offers={ Offers }
          currentCity={ City.COLOGNE }
          onCityClick={ emptyFunction }
          currentSorting={ SortList.PRICE_HIGH_TO_LOW }
          onSortingClick={ emptyFunction }
        />
    );
    expect(tree).toMatchSnapshot();
  });

  it(`without offers`, () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(
        <Main
          offers={ [] }
          currentCity={ City.COLOGNE }
          onCityClick={ emptyFunction }
          currentSorting={ SortList.TOP_RATED_FIRST }
          onSortingClick={ emptyFunction }
        />
    );
    expect(tree).toMatchSnapshot();
  });
});
