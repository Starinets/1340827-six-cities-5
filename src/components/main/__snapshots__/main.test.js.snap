// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component rendered correctly -> Main with offers 1`] = `
<div
  className="page page--gray page--main"
>
  <Memo(Connect(Header)) />
  <main
    className="page__main page__main--index "
  >
    <h1
      className="visually-hidden"
    >
      Cities
    </h1>
    <CityList
      currentCity="Cologne"
      onCityClick={[Function]}
    />
    <MainContent
      cityOffers={
        Array [
          Object {
            "adultsCount": 4,
            "bedroomsCount": 3,
            "city": Object {
              "location": Object {
                "latitude": 50.938361,
                "longitude": 6.959974,
                "zoom": 13,
              },
              "name": "Cologne",
            },
            "description": "Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.",
            "features": Array [
              "Laptop friendly workspace",
              "Washer",
              "Towels",
              "Fridge",
              "Air conditioning",
              "Baby seat",
              "Breakfast",
            ],
            "host": Object {
              "avatar": "img/avatar-angelina.jpg",
              "id": 25,
              "isPro": true,
              "name": "Angelina",
            },
            "id": 1,
            "image": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg",
            "images": Array [
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
            ],
            "isFavorite": false,
            "isPremium": false,
            "location": Object {
              "latitude": 50.957361,
              "longitude": 6.9509739999999995,
              "zoom": 16,
            },
            "name": "The house among olive ",
            "price": 493,
            "rating": 3.8,
            "type": "house",
          },
          Object {
            "adultsCount": 8,
            "bedroomsCount": 3,
            "city": Object {
              "location": Object {
                "latitude": 50.846557,
                "longitude": 4.351697,
                "zoom": 13,
              },
              "name": "Brussels",
            },
            "description": "This is a place for dreamers to reset, reflect, and create. Designed with a 'slow' pace in mind, our hope is that you enjoy every part of your stay; from making local coffee by drip in the morning, choosing the perfect record to put on as the sun sets.",
            "features": Array [
              "Towels",
              "Baby seat",
              "Laptop friendly workspace",
              "Dishwasher",
              "Washing machine",
              "Air conditioning",
              "Washer",
              "Coffee machine",
              "Breakfast",
              "Fridge",
            ],
            "host": Object {
              "avatar": "img/avatar-angelina.jpg",
              "id": 25,
              "isPro": true,
              "name": "Angelina",
            },
            "id": 2,
            "image": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg",
            "images": Array [
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/20.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg",
            ],
            "isFavorite": false,
            "isPremium": false,
            "location": Object {
              "latitude": 50.839557,
              "longitude": 4.346697,
              "zoom": 16,
            },
            "name": "Amazing and Extremely Central Flat",
            "price": 323,
            "rating": 4.3,
            "type": "apartment",
          },
          Object {
            "adultsCount": 7,
            "bedroomsCount": 4,
            "city": Object {
              "location": Object {
                "latitude": 50.846557,
                "longitude": 4.351697,
                "zoom": 13,
              },
              "name": "Brussels",
            },
            "description": "Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.",
            "features": Array [
              "Washer",
              "Laptop friendly workspace",
              "Breakfast",
              "Air conditioning",
              "Baby seat",
            ],
            "host": Object {
              "avatar": "img/avatar-angelina.jpg",
              "id": 25,
              "isPro": true,
              "name": "Angelina",
            },
            "id": 3,
            "image": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg",
            "images": Array [
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/12.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg",
            ],
            "isFavorite": false,
            "isPremium": false,
            "location": Object {
              "latitude": 50.865556999999995,
              "longitude": 4.371696999999999,
              "zoom": 16,
            },
            "name": "The house among olive ",
            "price": 218,
            "rating": 4.6,
            "type": "apartment",
          },
          Object {
            "adultsCount": 7,
            "bedroomsCount": 3,
            "city": Object {
              "location": Object {
                "latitude": 52.37454,
                "longitude": 4.897976,
                "zoom": 13,
              },
              "name": "Amsterdam",
            },
            "description": "Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.",
            "features": Array [
              "Dishwasher",
              "Air conditioning",
              "Laptop friendly workspace",
              "Washer",
              "Baby seat",
              "Towels",
              "Breakfast",
              "Coffee machine",
              "Fridge",
            ],
            "host": Object {
              "avatar": "img/avatar-angelina.jpg",
              "id": 25,
              "isPro": true,
              "name": "Angelina",
            },
            "id": 4,
            "image": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg",
            "images": Array [
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/12.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg",
            ],
            "isFavorite": false,
            "isPremium": false,
            "location": Object {
              "latitude": 52.385540000000006,
              "longitude": 4.902976,
              "zoom": 16,
            },
            "name": "Perfectly located Castro",
            "price": 440,
            "rating": 4.6,
            "type": "house",
          },
        ]
      }
      currentCity="Cologne"
      currentSorting="Price: high to low"
      onSortingClick={[Function]}
    />
  </main>
</div>
`;

exports[`Component rendered correctly -> Main without offers 1`] = `
<div
  className="page page--gray page--main"
>
  <Memo(Connect(Header)) />
  <main
    className="page__main page__main--index page__main--index-empty"
  >
    <h1
      className="visually-hidden"
    >
      Cities
    </h1>
    <CityList
      currentCity="Cologne"
      onCityClick={[Function]}
    />
    <MainEmpty
      currentCity="Cologne"
    />
  </main>
</div>
`;
